src/lib.rs

```diff
@@ -404,3 +404,10 @@ pub fn getrandom_uninit(dest: &mut [MaybeUninit<u8>]) -> Result<&mut [u8], Error
     // since it returned `Ok`.
     Ok(unsafe { slice_assume_init_mut(dest) })
 }
+
+// Don't run normal unit tests when testing custom getrandom
+#[cfg(all(
+    test,
+    not(all(target_family = "wasm", target_os = "unknown", feature = "custom"))
+))]
+pub(crate) mod tests;
```

Comment:
I think the "normal" tests should still be defined in tests/normal.rs (See comment below).

I think we're all keen to avoid duplicating complex `cfg` all over. I suggest we add a new api:
```
/// true if the implementation defined by `register_custom_getrandom!` would be used; false if it will be ignored.
pub const CUSTOM_USED: bool;
```
Then we can condition all the entropy checks in the common tests on that new API.

I think this API would also be useful to external users.

I also expected to see here something like:
```
cfg_if::cfg_if! {
    if #[cfg(any(target_arch = "x86", target_arch = "x86_64")] {
        mod rdrand;
        #[cfg(test)]
        mod rdrand_tests {
            tests::define_tests!(rdrand::getrandom_uninit);
        }
    }
}
```
This way, tests/rdrand.rs is properly replaced instead of just being lost.

Then presumably in future PRs we would do the same for use_file, etc.

Reply:
> I think we're all keen to avoid duplicating complex `cfg` all over. I suggest we add a new api:

After rearranging some stuff, this `cfg` is no longer needed. We can just do `#[cfg(test)]` and everything works as expected. Note that the CI tests are updated to run `wasm-pack test --node --features=custom --test=custom` when testing on an unsupported platform. This makes more sense to me, as the in-library unit tests will not build if you just enable the `custom` feature on an unsupported platform (you have to register a custom handler, which is only done in our integration test).



> I also expected to see here something like:
> 
> ```
> cfg_if::cfg_if! {
>     if #[cfg(any(target_arch = "x86", target_arch = "x86_64")] {
>         mod rdrand;
>         #[cfg(test)]
>         mod rdrand_tests {
>             tests::define_tests!(rdrand::getrandom_uninit);
>         }
>     }
> }
> ```
>
> This way, tests/rdrand.rs is properly replaced instead of just being lost.

I changed this PR to just unconditionally have `mod rdrand` present if the `"rdrand"` feature is enabled. This is similar to what we do for `"custom"`, and allows us to just have normal unit tests inside `rdrand.rs`. We can't put such things inside the `cfg_if`, but having the `mod rdrand` outside of that macro ends up being more readable anyway.